llvm/test/CodeGen/CHERI-Generic/Inputs/atomic-rmw-cap-ptr.ll

```diff
@@ -0,0 +1,86 @@
+; Check that we can generate sensible code for atomic operations using capability pointers
+; https://github.com/CTSRD-CHERI/llvm-project/issues/470
+@IF-RISCV@; RUN: %generic_cheri_purecap_llc %s -o - -mattr=+f,+a | FileCheck %s --check-prefixes=PURECAP,PURECAP-ATOMICS
```

Comment:
As with the float tests before, please use a hard-float ABI for all these.

Reply:
I guess this needs a new substitution in lit since you can only pass -target-abi once.
Is there anything else that needs fixing other than this file?

Reply:
I'm not a huge fan of the substitutions for llc, but I see why they're being done for CHERI-Generic. I didn't spot anything else.